lib/service_actor/checks/nil_check.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+# Ensure your inputs and outputs are not nil by adding `allow_nil: false`.
+#
+# Example:
+#
+#   class CreateUser < Actor
+#     input :name, allow_nil: false
+#     output :user, allow_nil: false
+#   end
+#
+#   class CreateUser < Actor
+#     input :name,
+#           allow_nil: {
+#             is: false,
+#             message: (lambda do |origin:, input_key:, actor:|
+#               "The value \"#{input_key}\" cannot be empty"
+#             end)
+#           }
+#
+#     input :phone, allow_nil: { is: false, message: "Phone must be present" }
+#
+#     output :user,
+#             allow_nil: {
+#               is: false,
+#               message: (lambda do |origin:, input_key:, actor:|
+#                 "The value \"#{input_key}\" cannot be empty"
+#               end)
+#             }
+#   end
+class ServiceActor::Checks::NilCheck < ServiceActor::Checks::Base
+  DEFAULT_MESSAGE = lambda do |origin:, input_key:, actor:|
+    "The \"#{input_key}\" #{origin} on \"#{actor}\" does not allow nil values"
+  end
+
+  private_constant :DEFAULT_MESSAGE
+
+  def self.check( # rubocop:disable Metrics/ParameterLists
+    origin:,
+    input_key:,
+    input_options:,
+    actor:,
+    conditions:,
+    result:,
+    **
+  ) # do
+    new(
+      origin: origin,
+      input_key: input_key,
+      input_options: input_options,
+      actor: actor,
+      allow_nil: conditions,
+      value: result[input_key],
+    ).check
+  end
+
+  def initialize( # rubocop:disable Metrics/ParameterLists
+    origin:,
+    input_key:,
+    input_options:,
+    actor:,
+    allow_nil:,
+    value:
+  ) # do
+    super()
+
+    @origin = origin
+    @input_key = input_key
+    @input_options = input_options
+    @actor = actor
+    @allow_nil = allow_nil
+    @value = value
+  end
+
+  def check
+    return unless @value.nil?
+
+    allow_nil, message =
+      define_allow_nil_and_message_from(@input_options[:allow_nil])
+
+    return if allow_nil?(allow_nil)
+
+    add_argument_error(
+      message,
+      origin: @origin,
+      input_key: @input_key,
+      actor: @actor,
+    )
+  end
+
+  private
+
+  def define_allow_nil_and_message_from(allow_nil)
+    if allow_nil.is_a?(Hash)
+      allow_nil.values_at(:is, :message)
```

Comment:
One (second to) last comment to point out that we might want to defend against missing `:message` keys.